src/lib/net/crates/codec/src/net_types/length_prefixed_vec.rs

```diff
@@ -4,18 +4,26 @@ use crate::net_types::var_int::VarInt;
 use std::io::{Read, Write};
 use tokio::io::AsyncWrite;
 
-#[derive(Debug)]
+#[derive(Clone, Debug, Eq, PartialEq)]
 pub struct LengthPrefixedVec<T> {
-    pub length: VarInt,
     pub data: Vec<T>,
 }
 
 impl<T> LengthPrefixedVec<T> {
     pub fn new(data: Vec<T>) -> Self {
-        Self {
-            length: VarInt::new(data.len() as i32),
-            data,
-        }
+        Self { data }
+    }
+
+    pub fn push(&mut self, value: T) {
+        self.data.push(value);
+    }
+
+    pub fn pop(&mut self) -> Option<T> {
+        self.data.pop()
+    }
+
+    pub fn length(&self) -> usize {
+        self.data.len()
     }
 }
```

Comment:
You can impl Deref and DerefMut, and return underlying data. And get all methods from data's Vec

Reply:
> You can impl Deref and DerefMut, and return underlying data. And get all methods from data's Vec

Yeah I'll probably do that lol